spec/client/Rollup-Integration-guide.md

```diff
@@ -0,0 +1,221 @@
+  }
+}
+```
+
+### `updateStateOnMisbehaviour`
+
+The misbehaviour update is also dependent on the rollup architecture. In sovereign proof-of-stake chains, if the consensus rules are violated, there is often no fallback mechanism as the trust in the chain is completely destroyed without out-of-protocol social consensus restarting the chain with a new validator set. Thus, for sovereign chains, a client should simply be disabled upon receiving valid misbehaviour.
+
+Rollups on the other hand do have a fallback layer in the data availability and settlement layers. For example, the settlement layer can verify a block is invalid and simply remove it thus enforcing that blocks can keep proceeding with valid states as the settlement layer can continue removing invalid blocks from the chain history. Similarly, it's possible that the settlement layer has a mechanism to switch the sequencer if a block is proven invalid.
+
+Thus, `updateStateOnMisbehaviour` can be less strict for rollups and simply remove the fraudulent consensus state and wait for the resolution as specified by the rollup's consensus rules.
+
+```typescript
+function updateStateOnMisbehaviour(clientMessage: ClientMessage) {
+  // unmarshalling logic ommitted
+  misbehaviour = Misbehaviour(clientMessage)
+  misbehavingHeight = getHeight(misbehaviour)
+
+  // delete the fraudulent consensus state
+  deleteConsensusState(clientMessage.clientId, misbehavingHeight)
+
+  // its possible for the rollup client to do additional logic here
+  // e.g. verify the next sequencer chosen from settlement layer
+  // however this is highly specific to rollup architectures
+  // and is not necessary for all rollup architectures
+  // so it will not be modelled here.
+}
+```
+
+### Membership Verification Methods
+
+The parts of the client that rely on the data availability are encapsulated in verifying new rollup blocks. Thus, once they are already added to the client, they can be used for proof verification without reference to the underlying data availability layer. For optimistic rollups, the consensus state must exist in the client for the full fraud period before it can be used for proof verification.
+
+Since the rollup client is dependent on underlying clients: data availability client and settlement client, these must also not be frozen by misbehaviour in order for proof verification to proceed.
+
+```typescript
+function verifyMembership(
+  clientState: ClientState,
+  height: Height,
+  delayPeriodTime: uint64, // disabled
+  delayPeriodBlocks: uint64, // disabled
+  proof: CommitmentProof,
+  path: CommitmentPath,
+  value: bytes
+): Error {
+  // check conditional clients are still valid
+  rollupClient = getClient(clientState.DALayer)
+  settlementClient = getClient(clientState.settlementLayer) // may not exist for all rollups
+
+  assert(isActive(clientState))
+  assert(isActive(rollupClient))
+  assert(isActive(settlementClient))
```

Comment:
The `rollupClient` here is the `clientState` passed as an arg. i.e the execution client?

```suggestion
  daClient = getClient(clientState.DALayer)
  settlementClient = getClient(clientState.settlementLayer) // may not exist for all rollups

  assert(isActive(clientState))
  assert(isActive(daClient))
  assert(isActive(settlementClient))
```